File: libraryFinal/src/main/java/com/library/Controller/ProductRegistration.java
package com.library.Controller;

import java.util.Locale;
import java.util.Scanner;

import com.library.Model.*;
import com.library.Repository.*;

public class ProductRegistration {

    ListDb lDb = new ListDb();

    public void ebookReg(){
        Product prod = new Ebook();
        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);
        System.out.println("Please, register a new Ebook: ");
        System.out.println("What is the name of the Ebook?");
        String nm = sc.nextLine();
        prod.setProductName(nm);
        System.out.println("What is the name of the Author?");
        String authorName = sc.nextLine();
        prod.setAuthorName(authorName);
        System.out.println("Write the description: ");
        String dc = sc.nextLine();
        prod.setDescription(dc);
        System.out.println("What is the value of the Ebook?");
        double vl = sc.nextDouble();
        prod.setValue(vl);
        System.out.println("What is the Publish Company? ");
        String pcn = sc.nextLine();
        prod.setPublishingCompanyName(pcn);
        lDb.setListArray(prod);
    }

    public void magazineReg(){
        Product prod = new Magazine();
        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);
        System.out.println("Please, register a new Magazine: ");
        System.out.println("What is the name of the Magazine?");
        String nm = sc.nextLine();
        prod.setProductName(nm);
        System.out.println("Write the description: ");
        String dc = sc.nextLine();
        prod.setDescription(dc);
        System.out.println("What is the value of the Magazine?");
        double vl = sc.nextDouble();
        prod.setValue(vl);
        System.out.println("What is the Publish Company? ");
        String pcn = sc.nextLine();
        prod.setPublishingCompanyName(pcn);
        lDb.setListArray(prod);
    }

    public void miniBookReg(){
        Product prod = new MiniBook();
        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);
        System.out.println("Please, register a new  MiniBook: ");
        System.out.println("What is the name of the MiniBook?");
        String nm = sc.nextLine();
        prod.setProductName(nm);
        System.out.println("Write the description: ");
        String dc = sc.nextLine();
        prod.setDescription(dc);
        System.out.println("What is the value of the MiniBook?");
        double vl = sc.nextDouble();
        prod.setValue(vl);
        System.out.println("What is the Publish Company? ");
        String pcn = sc.nextLine();
        prod.setPublishingCompanyName(pcn);
        lDb.setListArray(prod);
    }

    public void physicalBookReg(){
        Product prod = new MiniBook();
        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);
        System.out.println("Please, register a new Physical Book: ");
        System.out.println("What is the name of the Physical Book? ");
        String nm = sc.nextLine();
        prod.setProductName(nm);
        System.out.println("Write the description: ");
        String dc = sc.nextLine();
        prod.setDescription(dc);
        System.out.println("What is the value of the Physical Book? ");
        double vl = sc.nextDouble();
        prod.setValue(vl);
        System.out.println("What is the Publish Company? ");
        String pcn = sc.nextLine();
        prod.setPublishingCompanyName(pcn);
        lDb.setListArray(prod);
    }

    public void productShow(){
        lDb.getListArray();
    }
}

File: libraryCSharp/Controller/ProductRegistration.cs
using System;
using libraryCSharp.Model;
using libraryCSharp.Repository;

namespace libraryCSharp.Controller
{
    public class ProductRegistration
    {
        private ListDb lDb;

        public ProductRegistration()
        {
            lDb = new ListDb();
        }
        public void ebookReg(){
            Product prod = new Ebook();
            
            Console.WriteLine("Please, register a new Ebook: ");
            Console.WriteLine("What is the name of the Ebook?");
            prod.productName = Console.ReadLine();
            
            Console.WriteLine("What is the name of the Author?");
            prod.author.name = Console.ReadLine();
            
            Console.WriteLine("Write the description: ");
            prod.description = Console.ReadLine();

            Console.WriteLine("What is the value of the Ebook?");
            prod.value = double.Parse(Console.ReadLine());

            lDb.setList(prod);
        }
        public void magazineReg()
        {
            Product prod = new Magazine();
            Console.WriteLine("Please, register a new Magazine: ");
            Console.WriteLine("What is the name of the Magazine?");
            prod.productName = Console.ReadLine();

            Console.WriteLine("Write the description: ");
            prod.description = Console.ReadLine();

            Console.WriteLine("What is the value of the Magazine?");
            prod.value = double.Parse(Console.ReadLine());

            lDb.setList(prod);
        }
        public void miniBookReg(){
            Product prod = new MiniBook();
            Console.WriteLine("Please, register a new MiniBook: ");
            Console.WriteLine("What is the name of the MiniBook?");
            prod.productName = Console.ReadLine();
            
            Console.WriteLine("What is the name of the Author?");
            prod.author.name = Console.ReadLine();
            
            Console.WriteLine("Write the description: ");
            prod.description = Console.ReadLine();

            Console.WriteLine("What is the value of the MiniBook?");
            prod.value = double.Parse(Console.ReadLine());

            lDb.setList(prod);
        }
        public void physicalBookReg(){
            Product prod = new MiniBook();
            Console.WriteLine("Please, register a new Book: ");
            Console.WriteLine("What is the name of the Book?");
            prod.productName = Console.ReadLine();
            
            Console.WriteLine("What is the name of the Author?");
            prod.author.name = Console.ReadLine();
            
            Console.WriteLine("Write the description: ");
            prod.description = Console.ReadLine();

            Console.WriteLine("What is the value of the Book?");
            prod.value = double.Parse(Console.ReadLine());

            lDb.setList(prod);
        }
        public void productShow(){
            lDb.getListArray();
        }
    }
}